### PR TITLE
Feature/add endpoint team permissions CY-1243

### PR DESCRIPTION
**Pull request by peavers**

### CODE REVIEW

File: src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala [hunk 1]
```diff
@@ -30,19 +30,42 @@ class TeamServices(client: BitbucketClient) {
...
     */
-  def getRepositoryUserPermissions(
+  def getTeamRepositoryUserPermissions(
       username: String,
       repositorySlug: String,
       pageRequest: Option[PageRequest] = None
   ): RequestResponse[Seq[UserPermission]] = {
-    val baseRequestUrl = s"$BaseUrl/$username/permissions/repositories/$repositorySlug"
+    val encodedUsername = URLEncoder.encode(username, "UTF-8")
+    val encodedRepositorySlug = URLEncoder.encode(username, "UTF-8")
```
Comment (machadoit) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala]: Nice copy pasta bro! :D

File: src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala [hunk 2]
```diff
@@ -30,19 +30,42 @@ class TeamServices(client: BitbucketClient) {
     client.execute(Request(s"$BaseUrl/$encodedUsername", classOf[Team]))
   }
 
+  /**
+    * Retrieve the permissions for all teams matching the supplied username.
+    *
+    * @param username The username or the UUID of the account surrounded by curly-braces
```
Comment (bmbferreira) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala]: you missed the pageRequest param on the docs

File: src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala [hunk 3]
```diff
@@ -30,19 +30,42 @@ class TeamServices(client: BitbucketClient) {
...
+      case Some(request) =>
+        client.executeWithCursor[TeamPermission](baseRequestUrl, request)
+      case None =>
+        client.executePaginated(Request(baseRequestUrl, classOf[Seq[TeamPermission]]))
+    }
+  }
+
   /**
     * Retrieve the permissions matching the supplied username and repositorySlug.
     *
```
Comment (bmbferreira) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala]: it's also missing here

File: src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala [hunk 4]
```diff
@@ -30,19 +30,42 @@ class TeamServices(client: BitbucketClient) {
...
     * @return A [[RequestResponse]] with the user permissions for the repository
     */
-  def getRepositoryUserPermissions(
+  def getTeamRepositoryUserPermissions(
       username: String,
       repositorySlug: String,
       pageRequest: Option[PageRequest] = None
   ): RequestResponse[Seq[UserPermission]] = {
-    val baseRequestUrl = s"$BaseUrl/$username/permissions/repositories/$repositorySlug"
+    val encodedUsername = URLEncoder.encode(username, "UTF-8")
```
Comment (bmbferreira) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala]: 🙇

**Pull request merged**